packages/botonic-plugin-ner/src/preprocess/input-generator.ts

```diff
@@ -0,0 +1,45 @@
+import { Codifier } from '@botonic/nlp/dist/preprocess/codifier'
+import {
+  PADDING_TOKEN,
+  UNKNOWN_TOKEN,
+} from '@botonic/nlp/dist/preprocess/constants'
+import { Preprocessor } from '@botonic/nlp/dist/preprocess/preprocessor'
+import { tensor } from '@tensorflow/tfjs-core/dist/ops/tensor'
+import { Tensor2D } from '@tensorflow/tfjs-core/dist/tensor'
+
+export class InputGenerator {
```

Comment:
this is SRP :100: 
However, for me a "generator" sounds like the initial source of data (normally used for test). Please correct me if in NLP people use this terminology.
Please review NLP literature and libraries, but what about this?:
* rename Codifier to Encoder (google finds 1M search results for "nlp encoder" and 33k for "nlp codifier"
* rename InputGenerator to NerEncoder
as usual appy the agreed refactors in a separate PR at the end of the PR chain

Reply:
I named the class `Codifier` because it has two methods. One for encoding and the other for decoding. I thought it might be helpful to split this class into two: `Encoder` and `Decoder`. Maybe Even 4: `Encoder`, `CategoricalEncoder`, `Decoder`, `CategoricalDecoder`. What do you tink about this? @dpinol